banzai_floyds/tests/test_extract.py

```diff
@@ -1,118 +1,19 @@
-    # normalize out the polynomial so it is close to 1
-    continuum_polynomial /= np.mean(continuum_polynomial(np.arange(3000.0, 12000.1, 0.1)))
-    for i in range(2):
-        slit_coordinates = y2d - orders.center(x2d)[i]
-        in_order = orders.data == i + 1
-        trace_center = profile_centers[i](wavelengths.data)
-        if flat_spectrum:
-            data[in_order] += flux_normalization * gauss(slit_coordinates[in_order], trace_center[in_order],
-                                                         profile_sigma)
-        else:
-            profile = gauss(slit_coordinates[in_order], trace_center[in_order], profile_sigma)
-            input_spectrum = flux_normalization * continuum_polynomial(wavelengths.data[in_order]) * profile
-            for input_line, strength, width in zip(input_lines, input_line_strengths, input_line_widths):
-                # add some random emission lines
-                input_spectrum += strength * gauss(wavelengths.data[in_order], input_line, width) * profile
-            data[in_order] += input_spectrum
-        if include_background:
-            sky_wavelengths = np.arange(2500.0, 12000.0, 0.1)
-            sky_spectrum = np.zeros_like(sky_wavelengths)
-            sky_spectrum += sky_continuum
-            for line in SKYLINE_LIST:
-                line_spread = gauss(sky_wavelengths, line['wavelength'], fwhm_to_sigma(line_widths[i]))
-                sky_spectrum += line['line_strength'] * line_spread * sky_normalization
-            # Make a slow illumination gradient to make sure things work even if the sky is not flat
-            illumination = 100 * gauss(slit_coordinates[in_order], 0.0, 48)
-            input_sky[in_order] = np.interp(wavelengths.data[in_order], sky_wavelengths, sky_spectrum) * illumination
-            data[in_order] += input_sky[in_order]
-    data = np.random.poisson(data.astype(int)).astype(float)
-    data += np.random.normal(0.0, read_noise, size=data.shape)
-    errors = np.sqrt(read_noise**2 + np.abs(data))
-
-    frame = FLOYDSObservationFrame([CCDData(data, fits.Header({}), uncertainty=errors)], 'foo.fits')
-    frame.input_profile_centers = profile_centers
-    frame.input_profile_width = profile_width
-    frame.wavelengths = wavelengths
-    frame.orders = orders
-    if include_background:
-        frame.input_sky = input_sky
-    if not flat_spectrum:
-        frame.input_spectrum_wavelengths = np.arange(3000.0, 12000.0, 0.1)
-        frame.input_spectrum = flux_normalization * continuum_polynomial(frame.input_spectrum_wavelengths)
-        for input_line, strength, width in zip(input_lines, input_line_strengths, input_line_widths):
-            # add some random emission lines
-            frame.input_spectrum += strength * gauss(frame.input_spectrum_wavelengths, input_line, width)
-    return frame
+from banzai_floyds.utils.fitting_utils import fwhm_to_sigma
 
 
 def test_wavelength_bins():
```

Comment:
Is it worth a quick doc string or two to explicitly describe each of these tests, or are we trusting out future selves to get everything we need from the test names?

Reply:
I think the test names are pretty descriptive.